File: Data Structures & Algorithms/top-k-elements-in-list/submission-27.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        count = [[] for _ in range(len(nums)+1)]
        fq = {}

        for n in nums:
            fq[n] = fq.get(n,0)+1
        
        for i, v in fq.items():
            count[v].append(i)

        res = [] * k

        for i in range(len(count)-1,-1,-1):
            for n in count[i]:
                res.append(n)
            if len(res) == k:
                return res

File: Data Structures & Algorithms/anagram-groups/submission-27.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        let res = {};

        for (let s of strs) {
            let count = Array.from({length: 26}, () => 0);
            for (let c of s) {
                count[c.charCodeAt(0)-'a'.charCodeAt(0)]++
            }
            if (!res[count]) {
                res[count] = []
                res[count].push(s)
            } else {
                res[count].push(s)
            }
        }
        return Object.values(res);
        
    }
}
